storage/database/db_manager.go

```diff
@@ -1885,6 +1843,26 @@ func (dbm *databaseManager) ReadPreimageFromOld(hash common.Hash) []byte {
 	return data
 }
 
+func (dbm *databaseManager) WriteTrieNode(hash common.Hash, node []byte) {
+	dbm.lockInMigration.RLock()
+	defer dbm.lockInMigration.RUnlock()
+
+	if dbm.inMigration {
+		if err := dbm.getDatabase(StateTrieMigrationDB).Put(TrieNodeKey(hash), node); err != nil {
+			logger.Crit("Failed to store trie node", "err", err)
+		}
+	}
+	if err := dbm.getDatabase(StateTrieDB).Put(TrieNodeKey(hash), node); err != nil {
+		logger.Crit("Failed to store trie node", "err", err)
+	}
```

Comment:
This new changes (two writes to two different DB) is the same behavior with the code `dstDb.GetMemDB().Put()`?

Reply:
The `GetMemDB()` with `NewMemoryDBManager()` is only used in testing, where any `getDatabase(*)` call returns the same MemDB. If we limit the discussion to testing, then yes, it is equivalent to `GetMemDB().Put()`.

Reply:
Thanks.